src/Polly.Core/Telemetry/ResilienceStrategyTelemetry.cs

```diff
@@ -60,6 +60,17 @@ public void Report<TArgs, TResult>(ResilienceEvent resilienceEvent, ResilienceCo
         Listener.Write<TResult, TArgs>(new(TelemetrySource, resilienceEvent, context, args, outcome));
     }
 
+    /// <summary>
+    /// Updates the <see cref="ExecutionRejectedException.TelemetrySource"/> property of the exception.
+    /// </summary>
+    /// <param name="exception">The exception to update.</param>
+    public void UpdateTelemetrySource(ExecutionRejectedException exception)
```

Comment:
We can even hide this initially with `[EditorBrowsable(EditorBrowsableState.Never)]` ?